packages/components/Callout/macos/Callout.swift

```diff
@@ -321,4 +378,30 @@ private class FlippedVisualEffectView: NSVisualEffectView {
 	}
 }
 
+
+/// NSEvent localMonitors are an _old_ API that requires you to do some  memory management yourself.
+/// You _must_ call `removeMonitor`exactly once per monitor created from. This helper class enforces
+/// that with the use of an extra boolean to track whether we have already allocated/released our monitor.
+/// https://developer.apple.com/documentation/appkit/nsevent/1533709-removemonitor
+private class GuardedEventMonitor: Any {
+	
+	private var isMonitorAllocated = false
```

Comment:
and we believe boolean is good rather than an actual counter because one callout view should only have monitor...so it should always be +1 -1 ...?

Reply:
Yes. Multiple monitors would mean multiple blocks executed, which seems bad. I also thought I read somewhere if you call `addLocalMonitorForEvents(matching mask: handler:)` with the same mask twice, you get the same monitor twice. I haven't been able to prove that in code though..

In another case, maybe there are two different types of NSEvents you're listening for (mouseDown, keyDown ?) at which point you could maybe have two monitors. We don't need that for our case.